ionoscloud/utils.go

```diff
@@ -196,18 +152,6 @@ func responseBody(resp *ionoscloud.APIResponse) string {
 	return ret
 }
 
-func setPropWithNilCheck(m map[string]interface{}, prop string, v interface{}) {
-
-	rVal := reflect.ValueOf(v)
-	if rVal.Kind() == reflect.Ptr {
-		if !rVal.IsNil() {
-			m[prop] = rVal.Elem().Interface()
-		}
-	} else {
-		m[prop] = v
-	}
-}
-
 func IsValidUUID(uuid string) bool {
```

Comment:
Is this used now, anywhere? Since we have validation on uuid from terraform

Reply:
Good point, some volume and server fields have this check, but we should probably replace with the terraform validation.

Reply:
After looking a little more, we do use the function internally to check if a field is in UUID format.